### ISSUE TITLE
Improvise logging for Channel connection timeout on platform code

### ISSUE THREAD

**Issue by tiholic** [ably/ably-flutter#19]
### Why?

I've been calling `Realtime~channels~get~attach` without calling `Realtime~connect` first
This iteratively tries to attach to the channel and on timeout, RealtimeChannelStatus will be set as suspended. I wasn't sure until @paddybyers asked me whether I connected to the `realtime` instance in the first place!

### Suggestion

Improvise `VERBOSE`/`INFO` logging on timeout and if realtime instance status is **not** connected to show:
```
attach timeout, realtime instance is not connected
retrying attach
```

NOTE: This enhancement must be done on platform code `ably-java` and `ably-cocoa`

**Comment by QuintinWillison** [comment]
This suggestion is one possible solution. There may be others (e.g. fail faster).

See [internal conversation](https://ably-real-time.slack.com/archives/C0178QGC093/p1595335540029700).

**Comment by ikbalkaya** [comment]
Hmm. I just went through conversation and there doesn't seem to be a decision on how to tackle this. Considering this is an issue rooting from the platform implementations I would say they should be handled there. WDYT @QuintinWillison ? If so I can create issues for corresponding platforms

**Comment by QuintinWillison** [comment]
@ikbalkaya yes, please create platform issues and link them back to this issue. This issue should remain open until those dependencies have been updated here.
